types/user.ts

```diff
@@ -323,19 +335,66 @@ export class User extends LazyObject {
     // if content is a user, remove their avatar & bio & displayname.
     else if (content instanceof User) {
       content.deleteAvatar();
-      content.setBio("My profile content got deleted by a moderator and I am so embarrassed.");
-      content.setDisplayName("Disappointment");
+      content.setBio(
+        'My profile content got deleted by a moderator and I am so embarrassed.'
+      );
+      content.setDisplayName('Disappointment');
     }
 
     // add a modHistory entry
-    const modHistoryDocRef = doc(db, 'modHistory');
-    return setDoc(modHistoryDocRef, {
-      userModerated: (await content.getAuthor()).username,
-      userEmail: (await content.getAuthor()).email,
-      mod: this.docRef,
-      modReason: modReason,
-      timestamp: serverTimestamp()
-    });
+    this.addModAction(
+      await content.getAuthor(),
+      this,
+      'Deleted reported content: ' + modReason
+    );
+  }
+
+  /** getDateMM_DD_YYYY
+   * @returns the current date in the format MM_DD_YYYY
+   */
+  public async getDateMM_DD_YYYY() {
+    const date = new Date();
+    const day = date.getDate().toString().padStart(2, '0');
+    const month = (date.getMonth() + 1).toString().padStart(2, '0');
+    const year = date.getFullYear().toString();
+    return `${month}_${day}_${year}`;
+  }
+
+  /** addModAction
+   * Add a modAction array entry to the current day's doc in modHistory.
+   * @remarks this function should not be called directly.
+   * @param userModerated: the user who was moderated
+   * @param mod: the user who moderated
+   * @param modReason: the reason for the moderation
+   */
+  public async addModAction(userModerated: User, mod: User, modReason: String) {
+    const documentId = await this.getDateMM_DD_YYYY();
+    const modHistoryDocRef = await doc(db, 'modHistory', documentId);
+    const modHistorySnap = await getDoc(modHistoryDocRef);
```

Comment:
I can clean this up if you don't want to, but there's a way to update a document if it exists else create it, with `setDoc(docRef, content, {merge: true})`